src/StateManager.js

```diff
@@ -0,0 +1,64 @@
+// @flow
+
+function updateBrowserHistoryStack(currentStepName, nextStepName) {
+  const currentUrl = window.location.href;
+  const basePath = currentUrl.includes('?')
+    ? currentUrl.split('?')[0]
+    : currentUrl;
+  const searchParams = new URLSearchParams(new URL(currentUrl).searchParams);
+  searchParams.set('step', nextStepName);
+
+  window.history.pushState({ activeStep: currentStepName }, '', currentUrl);
+  window.history.replaceState(
+    { activeStep: nextStepName },
+    '',
+    `${basePath}?${searchParams.toString()}`,
+  );
+}
+
+function getActiveStepFromUrl() {
+  const { href } = window.location;
+  const searchParams = new URLSearchParams(new URL(href).searchParams);
+  return searchParams.get('step');
+}
+
+export default class StateManager {
```

Comment:
I'm mainly curious. What's your reason to use a class based over a purely functional approach?

Reply:
I chose a class based approach as I felt the api looked cleaner when calling methods via `state.updateHistory(currentStep.name, previousStep.name)` as opposed to state.updateHistory('url', currentStep.name, previousStep.name).

I believe that this also allows us to write cleaner code if we were to support multiple modes for persisting state.

```
const urlState = new StateManager('url');
const localStorageState = new StateManager('localStorageState');
```

Reply:
I refactored the code into function based approach here: cf34821.
So I guess, the usage would look like:
```
const urlState = StateManager('url');
const localStorageState = StateManager('localStorageState');

urlState.func();
localStorageState.func();
```